res/skins/QMLDemo/OrientationToggleButton.qml

```diff
@@ -3,78 +3,51 @@ import Mixxx 0.1 as Mixxx
 import QtQuick 2.12
 import QtQuick.Controls 2.12
 
-AbstractButton {
+Item {
     id: root
 
     property string group // required
     property string key // required
-    property alias orientation: control.value
+    property alias orientation: orientationSlider.value
     property color color: "white"
 
     implicitWidth: 56
     implicitHeight: 26
-    onClicked: control.value = Math.trunc(3 * pressX / root.width)
-    states: [
-        State {
-            name: "left"
-            when: orientation == 0
-
-            PropertyChanges {
-                target: indicator
-                x: 0
-            }
-
-        },
-        State {
-            name: "right"
-            when: orientation == 2
-
-            PropertyChanges {
-                target: indicator
-                x: parent.width - width
-            }
-
-        },
-        State {
-            name: "mid"
-            when: orientation == 1
-
-            PropertyChanges {
-                target: indicator
-                x: parent.width / 2 - width / 2
-            }
-
-        }
-    ]
-
-    Mixxx.ControlProxy {
-        id: control
 
-        group: root.group
-        key: root.key
-    }
+    Slider {
+        id: orientationSlider
 
-    Item {
-        anchors.fill: root
+        anchors.fill: parent
         anchors.leftMargin: 10
         anchors.rightMargin: 10
         anchors.topMargin: 2
         anchors.bottomMargin: 2
+        wheelEnabled: false
+        live: false
+        from: 0
+        to: 2
+        stepSize: 1
+        value: control.value
+        orientation: Qt.Horizontal
+        snapMode: Slider.SnapOnRelease
```

Comment:
Obviously, you can't add `value: orientationSlider.value` to the control proxy because that would create an endless binding loop. 

Instead, you only want to update the CO value when the slider was moved by the user, not on every value change.

Try this:

```suggestion
        snapMode: Slider.SnapOnRelease
        onMoved: control.value = value
```
Didn't test this, but it should work.

Reply:
Thanks, I thought I had figured it out but it seems like I didn't. Not sure onMoved will work because it might fire before releasing the control, but I'll try to figure something out.

Reply:
Turns out using onMoved causes some weird issues where the visualposition and the value of the slider get out of sync. I'll have to think of another workaround...

Reply:
I investigated a bit (using `console.warn(value)` :wink:), and this should work.

```suggestion
        snapMode: Slider.SnapOnRelease
        onMoved: {
            // The slider's `value` is not updated until after the move ended,
            // so we need to calculate the value position ourselves here.
            control.value = Math.round(visualPosition * (to - from));
        }
```

Reply:
Thanks. I don't know when I'll be able to get back on this though.

Reply:
Should work now.